sycl/source/detail/platform_impl.cpp

```diff
@@ -26,6 +27,50 @@ namespace detail {
 
 using PlatformImplPtr = std::shared_ptr<platform_impl>;
 
+// Return true if the given device 'Dev' matches with any filter
+bool isForcedDevice(detail::device_filter_list *FilterList,
+                    const plugin &Plugin, const RT::PiDevice &Dev,
+                    unsigned int Index) {
+  backend Backend = Plugin.getBackend();
+  info::device_type Type;
+  Plugin.call<PiApiKind::piDeviceGetInfo>(
+      Dev, pi::cast<RT::PiDeviceInfo>(info::device::device_type), sizeof(Type),
+      &Type, nullptr);
+
+  // According to SYCL spec host device must be always available
+  if (Type == info::device_type::host)
+    return true;
```

Comment:
Do we need to count the 'host' device as forced device by default?
The main purpose of this function is to give a bonus point to devices that match with SYCL_DEVICE_FILTER triples.
Unless 'host' device (or 'all') is specified in the SYCL_DEVICE_FILTER triple explicitly, 'host' is just the last device if there is no device is matched. 
So, counting the 'host' device automatically as the forced device seems not right because it will give extra 1000 points to the host device always.

Reply:
Host device is not treated as forced by default.